app/code/Magento/MediaContent/etc/di.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
+    <preference for="Magento\MediaContentApi\Api\AssignAssetInterface" type="Magento\MediaContent\Model\AssignAsset"/>
+    <preference for="Magento\MediaContentApi\Api\UnassignAssetInterface" type="Magento\MediaContent\Model\UnassignAsset"/>
+    <preference for="Magento\MediaContentApi\Api\GetAssetsUsedInContentInterface" type="Magento\MediaContent\Model\GetAssetsUsedInContent"/>
+    <preference for="Magento\MediaContentApi\Api\GetContentWithAssetInterface" type="Magento\MediaContent\Model\GetContentWithAsset"/>
+    <preference for="Magento\MediaContentApi\Api\ExtractAssetFromContentInterface" type="Magento\MediaContent\Model\ExtractAssetFromContent"/>
```

Comment:
Why interface and implementation located in one module and marked as API? It does not make sense for me. It would be better to leave the interface only. 

Also, I have noticed that such changes affect the install process, right now it's broken since some logic in the `Magento\Cms\Setup\Patch\Data\CreateDefaultPages` trying to use `Magento\MediaContentApi\Api\ExtractAssetFromContentInterface` during the install but config for that should be defined in the etc/di.xml.

 In the current implementation, it's not possible to move it to etc/di.xml since implementation should be configured from the other modules, but their configuration not available on the install step.

Reply:
The interface is used for injecting the service, the model is used for di configuration

That change was requested by @naydav at the online review